schematic/__init__.py

```diff
@@ -26,52 +36,116 @@
 logger = logging.getLogger(__name__)
 
 
-# borrowed from: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/tests/integration/conftest.py
-class FileSpanExporter(ConsoleSpanExporter):
-    """Create an exporter for OTEL data to a file."""
+def create_telemetry_session() -> requests.Session:
+    """
+    Create a requests session with authorization enabled if environment variables are set.
+    If no environment variables are set, the session will be created without authorization.
+
+    Returns:
+        requests.Session: A session object with authorization enabled if environment
+            variables are set. If no environment variables are set, the session will be
+            created without authorization. If no telemetry export format is set, None
+            will be returned.
+    """
+    tracing_export = os.environ.get("TRACING_EXPORT_FORMAT", None)
+    logging_export = os.environ.get("LOGGING_EXPORT_FORMAT", None)
+    if not (tracing_export or logging_export):
+        return None
+
+    session = requests.Session()
+    static_otlp_headers = os.environ.get("OTEL_EXPORTER_OTLP_HEADERS", None)
+    if static_otlp_headers:
+        logger.info(
+            "Using static OTLP headers set in environment variable `OTEL_EXPORTER_OTLP_HEADERS`."
+        )
+        return session
+
+    client_id = os.environ.get("TELEMETRY_EXPORTER_CLIENT_ID", None)
+    client_secret = os.environ.get("TELEMETRY_EXPORTER_CLIENT_SECRET", None)
+    client_token_endpoint = os.environ.get(
+        "TELEMETRY_EXPORTER_CLIENT_TOKEN_ENDPOINT", None
+    )
+    client_audience = os.environ.get("TELEMETRY_EXPORTER_CLIENT_AUDIENCE", None)
```

Comment:
I wrote this before we were 100% certain we were going to have an opentelemetry collector deployed to the schematic VPC where schematic API is running.

During normal operations we probably won't be using this OAuth2Client logic, however, if we ever need to configure the application to work like this it will be available for us to use. It also helped me to prove out the external telemetry system that is receiving these logs.